web/src/condense.js

```diff
@@ -218,6 +220,16 @@ export function condense_and_collapse(elems) {
         }
 
         const message_height = get_message_height(elem, message.id);
+
+        rows_to_resize.push({
+            elem,
+            $content,
+            message,
+            message_height,
+        });
+    }
+
+    for (const {elem, $content, message, message_height} of rows_to_resize) {
```

Comment:
Let's add a comment here that we do this in a separate loop to avoid layout thrashing; this is the kind of thing where someone might "clean up" in a refactoring absent a comment explaining why we do two loops.

Reply:
Added a comment